pkg/controllers/user/authz/podsecuritypolicy/template.go

```diff
@@ -66,19 +66,20 @@ func (m *templateManager) sync(key string, obj *v3.PodSecurityPolicyTemplate) er
 }
 func fromTemplate(policies v1beta12.PodSecurityPolicyInterface, policyLister v1beta12.PodSecurityPolicyLister,
 	key string, originalTemplate *v3.PodSecurityPolicyTemplate) (*v1beta1.PodSecurityPolicy, error) {
-	return fromTemplateExplicitName(policies, policyLister, keyToPolicyName(key), originalTemplate)
+	return fromTemplateExplicitName(policies, policyLister, keyToPolicyName(key), originalTemplate, key)
```

Comment:
You shouldnt need this KeyToPolicyName function. Or, at least, you shouldnt need to use key. the originalTemplate.Name should have the ID you need